### PR TITLE
refactor: enable more eslint rules

### PR DESCRIPTION
**Pull request by pitaj**

### CODE REVIEW

**Comment by pitaj** [comment]
Ah node 10 doesn't have flatMap

**Comment by julianlam** [comment]
Just checking that the arrow-function enforcing doesn't cause issues if you write

``` js
function fooBar () {
  return this._var;
}
```

?

**Review by julianlam — APPROVED** [review]
I honestly have no problem with any of these changes. It's mostly enforcing use of `let`/`const`, arrow functions, (I think) hoisting for organizational purposes, template strings...

If you have eslint installed in vscode, then this should all work without issue and auto-fix on save (if enabled, which I recommend)

**Comment by julianlam** [comment]
If @psychobunny and @barisusakli have no issues, this LGTM

File: src/controllers/top.js
Comment (barisusakli) [inline src/controllers/top.js:22]: Should be `const [, feedQs] = data.rssFeedUrl.split('?');` ?

Comment (barisusakli) [reply src/controllers/top.js:22]: `eslint prefer-destructuring` Im not sold on, can lead to bugs, I think previous version is more readable

**Comment by barisusakli** [comment]
Can we hold off on merging this until categories-refactor is merged, dont want this to conflict with everything there.

File: src/controllers/top.js
Comment (pitaj) [reply src/controllers/top.js:22]: I can disable it for arrays

**Comment by pitaj** [comment]
> Just checking that the arrow-function enforcing doesn't cause issues if you write
> 
> ```js
> function fooBar () {
>   return this._var;
> }
> ```

It does not affect functions which use `this`

**Review by barisusakli — APPROVED** [review]

**Comment by julianlam** [comment]
All lines now show Peter's name on git blame. Reassign all bugs to him. :shipit:

**Pull request merged**
